rest/load-rest-resources.ts

```diff
@@ -24,9 +25,26 @@ export function loadRestResources({
 
   return Object.fromEntries(
     Object.entries(resources).map(([name, resource]) => {
-      class NewResource extends resource {
-        public static CLIENT = RestClient;
-      }
+      class NewResource extends resource {}
+
+      NewResource.setClassProperties({
+        CLIENT: RestClient,
+        CONFIG: config,
+      });
+
+      Object.entries(NewResource.HAS_ONE).map(([_attribute, klass]) => {
+        (klass as typeof Base).setClassProperties({
+          CLIENT: RestClient,
+          CONFIG: config,
+        });
+      });
+
+      Object.entries(NewResource.HAS_MANY).map(([_attribute, klass]) => {
+        (klass as typeof Base).setClassProperties({
+          CLIENT: RestClient,
+          CONFIG: config,
+        });
+      });
```

Comment:
Any this also fixes any other requests that depend on a client, like

```ts
product.variants[1].save();
```

which would fail due to the lack of a client before.